### DISCUSSION TITLE
Migrate existing react app with components to a mono repo with shared components

### DISCUSSION THREAD

**Discussion by JordanDDisch** [nrwl/nx#10384] [General]
We currently have an exiting CRA that has a bunch of components. We would like to migrate to a mono repo to support both a web and native app with shared components. How do we get the typescript configuration to build the existing component library and also slowly migrate components to the shared library folder. Imagine this is a common situation for some people.

Having issues keeping our absolute paths in the existing app and getting the shared component paths to work together in the CRA app.

**Comment by rfprod** [comment]
Have you tried the following approach?
1. [create a workspace](https://nx.dev/react-tutorial/01-create-application#create-a-new-workspace) `npx create-nx-workspace@latest`
2. put your monolith sources into the generated app 
2.1. fix imports as needed, absolute paths **should not** be used, it should be relative to your monorepo application root directory at first, [this vscode plugin](https://marketplace.visualstudio.com/items?itemName=pmneo.tsimporter) may help
2.2 make sure the app works as expected after migrating sources
4. migrate the monolith gradually
3.1 plan migration (how to split monolith into libraries)
3.2 [generate](https://nx.dev/cli/generate) libraries as needed `nx generate @nrwl/react:library my-library`
3.3 move sources from your monolithic app to libs and update imports in your app

If you don't want to replace imports manually and your app sources are not messy, `find` + [`sed`](https://www.gnu.org/software/sed/manual/sed.html) may be an option to batch replace absolute paths

```bash
find apps/myapp -type f -name "*.ts" -exec sed -i 's/foo/bar/g' {} +
```

**Reply by JordanDDisch** [reply]
@rfprod cheers for the response. Not married to the idea of keeping our absolute paths, but was wondering if anyones had this issue before and had a tsconfig ready for copy paste XD. For me it's about keeping the PR diff small, but ill defo have a go at the approach above though and see if that works. Thanks for the nice command for find replace!

Have you had to do this before?

**Reply by rfprod** [reply]
@JordanDDisch I have migrated Angular apps using this approach because I work primarily with Angular, but the process is generic and should work for React as well.
